sympy/functions/elementary/exponential.py

```diff
@@ -762,10 +762,15 @@ def _eval_expand_log(self, deep=True, **hints):
             return expand_log(self.func(*self.args), deep=deep, force=force)
         arg = self.args[0]
         if arg.is_Integer:
-            # expand log as product of its prime factors
-            p = factorint(int(arg))
-            if int(arg) not in p.keys() and factor:
-                return sum(n*log(val) for val, n in p.items())
+            # remove perfect powers
```

Comment:
```
            p = perfect_power(int(arg))
            logarg = None
            coeff = 1
            if p is not False:
                arg, coeff = p
                logarg = self.func(arg)
            # expand as product of its prime factors if factor=True
            if factor:
                p = factorint(int(arg))
                if int(arg) not in p.keys():
                    logarg = sum(n*log(val) for val, n in p.items())
            if logarg is not None:
                return coeff*logarg
```

Reply:
I will make this change.

Reply:
Having second thoughts...you could just check the for the `coeff = igcd(p.values())` and if it isn't 1 then all values can be divided by `coeff`.

Reply:
I didn't understand the use of `igcd(p[0],p[1])`, 
if p = (3,3), then `log(27)/3` wont be solved further